awx/main/tasks.py

```diff
@@ -952,184 +954,247 @@ def run(self, pk, **kwargs):
         '''
         Run the job/task and capture its output.
         '''
-        instance = self.update_model(pk, status='running',
-                                     start_args='')  # blank field to remove encrypted passwords
+        self.instance = self.update_model(pk, status='running',
+                                          start_args='')  # blank field to remove encrypted passwords
 
-        instance.websocket_emit_status("running")
+        self.instance.websocket_emit_status("running")
         status, rc, tb = 'error', None, ''
         output_replacements = []
         extra_update_fields = {}
-        event_ct = 0
         stdout_handle = None
+        fact_modification_times = {}
+        self.event_ct = 0
+        private_data_dir = None
 
         try:
-            kwargs['isolated'] = instance.is_isolated()
-            self.pre_run_hook(instance, **kwargs)
-            if instance.cancel_flag:
-                instance = self.update_model(instance.pk, status='canceled')
-            if instance.status != 'running':
+            isolated = self.instance.is_isolated()
+            self.pre_run_hook(self.instance)
+            if self.instance.cancel_flag:
+                self.instance = self.update_model(self.instance.pk, status='canceled')
+            if self.instance.status != 'running':
                 # Stop the task chain and prevent starting the job if it has
                 # already been canceled.
-                instance = self.update_model(pk)
-                status = instance.status
-                raise RuntimeError('not starting %s task' % instance.status)
+                self.instance = self.update_model(pk)
+                status = self.instance.status
+                raise RuntimeError('not starting %s task' % self.instance.status)
 
             if not os.path.exists(settings.AWX_PROOT_BASE_PATH):
                 raise RuntimeError('AWX_PROOT_BASE_PATH=%s does not exist' % settings.AWX_PROOT_BASE_PATH)
 
             # store a record of the venv used at runtime
-            if hasattr(instance, 'custom_virtualenv'):
-                self.update_model(pk, custom_virtualenv=getattr(instance, 'ansible_virtualenv_path', settings.ANSIBLE_VENV_PATH))
-
-            # Fetch ansible version once here to support version-dependent features.
-            kwargs['ansible_version'] = get_ansible_version()
```

Comment:
Is there any way we can leave the calculation of ansible_version here? because I would just be adding it back.

This seems like a reasonable thing that could be passed around, maybe `self.ansible_version` would be better?